### PR TITLE
[master < T0906-MG] different results returned when executing the same queries on memgraph and neo4j respectively

### PR DESCRIPTION
**Pull request by gvolfing**
[master < Task] PR
- [ ] Check, and update documentation if necessary - **Not needed.**
- [x] Update [changelog](https://docs.memgraph.com/memgraph/changelog)

Docs PR:
https://github.com/memgraph/docs/pull/479

### CODE REVIEW

**Comment by gvolfing** [comment]
I will also investigate the `count` function in this PR based on this [comment](https://github.com/memgraph/memgraph/issues/343#issuecomment-1041470777).

File: tests/gql_behave/tests/memgraph_V1/features/aggregations.feature [hunk 1]
```diff
@@ -237,6 +277,16 @@ Feature: Aggregations
...
+    Scenario: Max test 04:
+        Given an empty graph
+        When executing query:
+        """
+        RETURN max(null)
+        """
+        Then the result should be:
+            | max(null) |
+            | null      |
+
```
Comment (antaljanosbenjamin) [inline tests/gql_behave/tests/memgraph_V1/features/aggregations.feature:289]: I like the new test cases! First I thought I can construct a case which breaks the compatibility with openCypher/Neo4j:

```
memgraph> MATCH (n) DELETE n;
Empty set (0.001 sec)
1 nodes have been deleted.
memgraph> CREATE (n {property: 1});
Empty set (0.001 sec)
1 nodes have been created.
memgraph> CREATE (n);
Empty set (0.001 sec)
1 nodes have been created.
memgraph> MATCH (n) RETURN count(n) < n.property;
+-----------------------+
| count(n) < n.property |
+-----------------------+
| Null                  |
| false                 |
+-----------------------+
2 rows in set (0.001 sec)
```
But, fortunately I was shocked that is the exact result Neo4j returns. So this test is also passed. However I would introduce one more test that checks the case when the `group_by` expression sometimes gets evaluated to `Null`, something with the following queries:
```
CREATE (n);
CREATE (n {property: 1});
MATCH (n) RETURN count(n) < n.property, count(n.property), count(n), avg(n.property), min(n.property), max(n.property), sum(n.property);
```
And the expected result should be:
```
+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
| count(n) < n.property | count(n.property)     | count(n)              | avg(n.property)       | min(n.property)       | max(n.property)       | sum(n.property)       |
+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
| Null                  | 0                     | 1                     | Null                  | Null                  | Null                  | 0                     |
| false                 | 1                     | 1                     | 1                     | 1                     | 1                     | 1                     |
+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
```

File: tests/unit/query_plan_accumulate_aggregate.cpp [hunk 2]
```diff
@@ -217,31 +217,31 @@ TEST_F(QueryPlanAggregateOps, WithData) {
 TEST_F(QueryPlanAggregateOps, WithoutDataWithGroupBy) {
   {
     auto results = AggregationResults(true, {Aggregation::Op::COUNT});
-    EXPECT_EQ(results.size(), 0);
+    EXPECT_EQ(results.size(), 1);
```
Comment (antaljanosbenjamin) [inline tests/unit/query_plan_accumulate_aggregate.cpp:220]: Maybe, could you please check their type and value also?

File: tests/unit/query_plan_accumulate_aggregate.cpp [hunk 3]
```diff
@@ -257,13 +257,13 @@ TEST_F(QueryPlanAggregateOps, WithoutDataWithoutGroupBy) {
...
   EXPECT_TRUE(results[0][3].IsNull());
-  // sum
-  EXPECT_TRUE(results[0][4].IsNull());
-  // avg
+  //  sum
+  EXPECT_EQ(results[0][4].ValueInt(), 0);
+  //  avg
   EXPECT_TRUE(results[0][5].IsNull());
-  // collect list
+  //  collect list
```
Comment (antaljanosbenjamin) [inline tests/unit/query_plan_accumulate_aggregate.cpp:266]:
```suggestion
  // collect list
```

**Review by antaljanosbenjamin — CHANGES_REQUESTED** [review]

File: tests/gql_behave/tests/memgraph_V1/features/aggregations.feature (hunk 1 above)
Comment (gvolfing) [reply tests/gql_behave/tests/memgraph_V1/features/aggregations.feature:289]: I am thinking if the ordering of the rows might cause some problems, because for me its the other way around, so:
+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
                               │| count(n) < n.property | count(n.property)     | count(n)              | avg(n.property)       | min(n.property)       | max(n.property)       | sum(n.property)       |
                               │+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
                               │| false                 | 1                     | 1                     | 1                     | 1                     | 1                     | 1                     |
                               │| Null                  | 0                     | 1                     | Null                  | Null                  | Null                  | 0                     |
                               │+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+

**Review by jbajic — APPROVED** [review]

File: tests/gql_behave/tests/memgraph_V1/features/aggregations.feature (hunk 1 above)
Comment (antaljanosbenjamin) [reply tests/gql_behave/tests/memgraph_V1/features/aggregations.feature:289]: I copied it from Memgraph, but I might created the vertices in reversed order.

**Review by antaljanosbenjamin — APPROVED** [review]
